### DISCUSSION TITLE
Can Copilot read an existed git repo to post more comprehensive suggestions?

### DISCUSSION THREAD

**Discussion by wonter123** [community/community#51539] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi All, 
I am newbie to use copilot and I find it is amazing to help on coding! When establishing new codes it works really well! However, I have some questions for using copilot in large repository or one project folder in one big git clone repo. Will copilot understand codes and logics outside current file (if it has several dependencies on other classes, packages etc.)? Do I need to do any sorts of configuration for copilot to achieve this, reading the repo? Also does copilot have the ability to understand configuration files like json, xml or ini? Thanks!!

**Comment by pstepanek954** [comment]
Ajajah

**Comment by Akash1134** [comment, marked as answer]
Hi @wonter123 ,

We announced a feature "[Referencing public code](https://github.blog/2022-11-01-preview-referencing-public-code-in-github-copilot/)" back in November as a preview that we're still working on. With that in place, within this context, we will begin to introduce several new capabilities to Copilot. For any code fragment suggested by Copilot, developers should have:

- An inventory of similar code found in GitHub public repositories
- The ability to sort that inventory by repository license, commit date, etc.

There is another exciting work in progress that'll let GitHub Copilot see your entire repo when it comes up with its suggestions. Specifically, showing Copilot snippets from my private codebase that are relevant to the task you're trying to accomplish. For details, see [GitHub Copilot for *Your* Codebase](https://githubnext.com/projects/copilot-view/)

**Reply by wonter123** [reply]
Hi Akash,
Thanks for your response and really appreciate your explanation! So, for copilot today it basically understands a general context of current file and get you an answer based on its training data and current file's code style. And in the future, copilot will be able to find reference for its suggestions and will be able to understand the codebase in a more comprehensive way. Do I understand this correctly? Thanks again for the help!

**Reply by Akash1134** [reply]
Yes! You get that spot on @wonter123 🙂

**Reply by ritajie** [reply]
Thanks to Github for the effort, this is a fascinating feature and I can't wait to get to work with it 🤩

**Reply by sis-skakani** [reply]
Any update on this since March 31 ? as its been nearly 7-8 months  and no information yet

**Reply by Experilous** [reply]
They're planning on how high the prices can be kept

**Comment by orphansystem** [comment]
Simple explanation: Not yet.

**Reply by Experilous** [reply]
i must urge they never and only way to build one personally own and make opensource to help all
